auditlog/models/rule.py

```diff
@@ -307,23 +301,67 @@ def create_logs(self, uid, res_model, res_ids, method,
             diff = DictDiffer(
                 new_values.get(res_id, EMPTY_DICT),
                 old_values.get(res_id, EMPTY_DICT))
-            self._create_log_line_on_write(
-                log, diff.changed(), old_values, new_values)
-            self._create_log_line_on_create(log, diff.added(), new_values)
+            if method is 'create':
+                self._create_log_line_on_create(log, diff.added(), new_values)
+            elif method is 'read':
+                self._create_log_line_on_read(
+                    log, old_values.get(res_id, EMPTY_DICT).keys(), old_values)
+            elif method is 'write':
+                self._create_log_line_on_write(
+                    log, diff.changed(), old_values, new_values)
 
     def _get_field(self, model, field_name):
         cache = self.pool._auditlog_field_cache
         if field_name not in cache.get(model.model, {}):
             cache.setdefault(model.model, {})
-            # We use 'search()' then 'read()' instead of the 'search_read()'
-            # to take advantage of the 'classic_write' loading
+            # - we use 'search()' then 'read()' instead of the 'search_read()'
+            #   to take advantage of the 'classic_write' loading
+            # - search the field in the current model and those it inherits
             field_model = self.env['ir.model.fields']
+            all_model_ids = [model.id]
+            all_model_ids.extend(model.inherited_model_ids.ids)
             field = field_model.search(
-                [('model_id', '=', model.id), ('name', '=', field_name)])
-            field_data = field.read(load='_classic_write')[0]
-            cache[model.model][field_name] = field_data
+                [('model_id', 'in', all_model_ids), ('name', '=', field_name)])
+            # The field can be a dummy one, like 'in_group_X' on 'res.users'
+            # As such we can't log it (field_id is required to create a log)
+            if not field:
+                cache[model.model][field_name] = False
+            else:
+                field_data = field.read(load='_classic_write')[0]
```

Comment:
Another problem too about calls to ``read`` on ``ir.model.fields`` if audit ``read`` rules are defined on it (did not tested, but it should fail I think) : check if there is a ``read.origin`` attribute and call this one to avoid the recursivity.